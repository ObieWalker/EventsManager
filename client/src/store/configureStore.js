import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;