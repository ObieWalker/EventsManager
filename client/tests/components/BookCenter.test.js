import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { BookCenter }
  from '../../src/components/UserDashboard/BookCenterModal';

let mountedComponent;
let props;
const mockStore = configureStore();
// const createCenter = jest.fn(() => Promise.resolve({}));

let wrapper;
const createSuccess = '';
const createError = '';
const center = {
  name: 'random',
  address: 'random address',
  city: 'random',
  capacity: 500,
  facility: 'Tables and Chairs'
};
// const pickadate = jest.fn();
const hideModal = jest.fn();
const addNewEvent = jest.fn(() => Promise.resolve({}));
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      center,
      addNewEvent,
      createSuccess,
      createError,
      hideModal
    };
    mountedComponent = shallow(<BookCenter {...props} />);
  }
  return mountedComponent;
};

describe('Book Center Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().clear).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().handleChange).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().formIsValid).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().onSubmit).toBeDefined();
  });
  it('should have a method that handles handleDateChange', () => {
    expect(wrapper.instance().handleDateChange).toBeDefined();
  });
  it('testing book center function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().onSubmit(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });
//   it('testing clear function', () => {
//     wrapper = getComponent();
//     wrapper.instance().reset();
//     expect(wrapper.state().name).toEqual('');
//     expect(wrapper.state().address).toEqual('');
//     expect(wrapper.state().city).toEqual('');
//     expect(wrapper.state().capacity).toEqual('');
//     expect(wrapper.state().facility).toEqual('');
//     expect(wrapper.state().errors).toEqual({});
//   });
});

describe('Connected Component', () => {
  describe('Connected mountedLogin', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        center,
        addNewEvent,
        createSuccess,
        createError,
        hideModal
      });
      wrapper = shallow(<connect store={store}
        hideModal={hideModal}
        createSuccess={createSuccess}
        createError={createError}
        center={center}
        addNewEvent={addNewEvent}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});
