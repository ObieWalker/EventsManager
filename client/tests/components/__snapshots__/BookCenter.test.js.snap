// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Center Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookCenter
    addNewEvent={[MockFunction]}
    center={
      Object {
        "address": "random address",
        "capacity": 500,
        "city": "random",
        "facility": "Tables and Chairs",
        "name": "random",
      }
    }
    createError=""
    createSuccess=""
    hideModal={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "bookCenterValue": undefined,
      "bookDateValue": undefined,
      "bookOnSubmit": [Function],
      "center": Object {
        "address": "random address",
        "capacity": 500,
        "city": "random",
        "facility": "Tables and Chairs",
        "name": "random",
      },
      "centerName": "random",
      "dateOnChange": [Function],
      "errors": Object {},
      "guestValue": 100,
      "onAfterChange": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "sliderMax": 100000,
      "sliderOnChange": [Function],
      "sliderStep": 50,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "bookCenterValue": undefined,
        "bookDateValue": undefined,
        "bookOnSubmit": [Function],
        "center": Object {
          "address": "random address",
          "capacity": 500,
          "city": "random",
          "facility": "Tables and Chairs",
          "name": "random",
        },
        "centerName": "random",
        "dateOnChange": [Function],
        "errors": Object {},
        "guestValue": 100,
        "onAfterChange": [Function],
        "onChange": [Function],
        "onFocus": [Function],
        "sliderMax": 100000,
        "sliderOnChange": [Function],
        "sliderStep": 50,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
